### PR TITLE
Remove use of deprecated python distutils

### LINKED ISSUE #781: fix antelope-spring-dev.deb install with Python 3.12 (as on Ubuntu 24)
The install script for the dev packages makes use of distutils,
https://github.com/AntelopeIO/spring/blob/5fbe2011276c7acc34b672026119ff7103c9b16f/scripts/postinst#L4-L6
but distuilts is removed as of Python 3.12. Need to do something else here for proper Ubuntu 24 support

### PR DESCRIPTION
**Pull request by heifner**
Remove use of deprecated python `distutils`.

Resolves #781

### CODE REVIEW

**Comment by linh2931** [comment]
Failed for ubntu20?

File: scripts/prerm [hunk 1]
```diff
@@ -1,6 +1,15 @@
-#!/bin/sh
+:#!/bin/sh
```
Comment (greg7mdp) [inline scripts/prerm:1]: this seems like a typo.

**Review by greg7mdp — APPROVED** [review]

File: scripts/prerm (hunk 1 above)
Comment (heifner) [reply scripts/prerm]: Yep, thanks.

File: scripts/postinst [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
 
 set -e
-Python_SITELIB=$(python3 -c "from distutils import sysconfig;print(sysconfig.get_python_lib(plat_specific=False,standard_lib=False))")
+Python_SITELIB=$(python3 <<EOF
+import sys
+if sys.version_info < (3, 10, 0):
+    from distutils import sysconfig
+    print(sysconfig.get_python_lib(plat_specific=False,standard_lib=False))
+else:
+    from sysconfig import get_path; print(get_path('platlib', 'deb_system'))
```
Comment (spoonincode) [inline scripts/postinst:10]: Do you know where the `deb_system` scheme is documented?

**Review by spoonincode — APPROVED** [review]

File: scripts/postinst (hunk 2 above)
Comment (heifner) [reply scripts/postinst:10]: No :(
Here is where I got that from: https://forum.freecad.org/viewtopic.php?t=67985
Which links to this "interesting" article: https://ffy00.github.io/blog/02-python-debian-and-the-install-locations/
Add this to the many things I hate about Python.

**Review by linh2931 — APPROVED** [review]

**Comment by ericpassmore** [comment]
Note:start
category: Chores
component: Internal
summary: Removed use of deprecated python distutils.
Note:end

**Pull request merged**